tests/PySys/log_request_end_to_end/log_request_num_lines/run.py

```diff
@@ -0,0 +1,87 @@
+import sys
+import time
+import os
+import subprocess
+import requests
+
+"""
+Validate end to end behaviour for the log request operation.
+
+When we send a log request ( for all text with, 250 lines) from cumulocity to device and wait for some time.
+Then sm mapper receives the request and sends the response
+Validate if the response contains the log file for number of lines.
+If number of lines are greater than 250 then pass the test
+Else stop and cleanup the operation by sending operation failed message.
+"""
+
+from environment_c8y import EnvironmentC8y
+
+
+class LogRequestVerifyNumberOfLines(EnvironmentC8y):
+
+    def setup(self):
+        super().setup()
+        self.create_logs_for_test()
+        self.addCleanupFunction(self.cleanup_logs)
+
+    def execute(self):
+        log_file_request_payload = {
+            "dateFrom": "2021-11-15T18:55:49+0530",
+            "dateTo": "2021-11-21T18:55:49+0530",
+            "logFile": "software-management",
+            "searchText": "",
+            "maximumLines": 250
+        }
+        self.cumulocity.trigger_log_request(log_file_request_payload)
+
+    def validate(self):
+        status = self.wait_until_retrieved_logs()
+        if not status:
+            self.log.info("Explicitly send the operation status as failed.")
```

Comment:
If I put it in the cleanup, it will be called always either the test passes or fails. Here it should be called only when the test fails to retrieve logs.